Library/Homebrew/tap.rb

```diff
@@ -280,4 +317,75 @@ def self.each
   def self.names
     map(&:name)
   end
+
+  private
+
+  # @private
+  def formula_file_to_name(file)
+    "#{name}/#{file.basename(".rb")}"
+  end
+
+  # @private
+  def alias_file_to_name(file)
+    "#{name}/#{file.basename}"
+  end
+end
+
+class CoreFormulaTap < Tap
+  # @private
+  def initialize
+    @user = "Homebrew"
+    @repo = "homebrew"
+    @name = "Homebrew/homebrew"
+    @path = HOMEBREW_REPOSITORY
+  end
+
+  include Singleton
+
+  undef_method :install
+  undef_method :uninstall
+  undef_method :command_files
+  undef_method :pinned_symlink_path
+  undef_method :pinned?
+  undef_method :pin
+  undef_method :unpin
+  undef_method :to_hash
```

Comment:
> this shows more that this is not a "real" Tap.

yet. It will become .

In fact the only methods I need to block are `uninstall`, `pin` and `unpin`. (`install` will be preserved  for future use.) So how about I overwrite them to no-op. like:
```ruby
def uninstall
  raise "Tap#uninstall is not available for CoreFormulaTap"
end

def pin
  raise "Tap#pin is not available for CoreFormulaTap"
end

def unpin
  raise "Tap#unpin is not available for CoreFormulaTap"
end

def pinned?
  false
end
````